presto-main/src/main/java/com/facebook/presto/operator/aggregation/noisyaggregation/NoisySumAggregationUtils.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.noisyaggregation;
+
+import com.facebook.presto.common.block.BlockBuilder;
+import com.facebook.presto.spi.PrestoException;
+
+import java.security.SecureRandom;
+import java.util.Random;
+
+import static com.facebook.presto.common.type.DoubleType.DOUBLE;
+import static com.facebook.presto.spi.StandardErrorCode.INVALID_FUNCTION_ARGUMENT;
+
+public class NoisySumAggregationUtils
+{
+    private NoisySumAggregationUtils()
+    {
+    }
+
+    public static void updateNoisySumState(NoisySumState state, double value, double noiseScale, Double lower, Double upper, Long randomSeed)
```

Comment:
Should this be defined as an instance method in `NoisySumState` instead?

Reply:
I dont think it should be. `NoisySumState` is only an interface, not a class though.

Reply:
Good point. My bad!